pkgs/development/embedded/bootcommander/default.nix

```diff
@@ -0,0 +1,32 @@
+{ lib, stdenv, fetchgit, cmake, libopenblt }:
+
+stdenv.mkDerivation rec {
+
+  pname = "bootcommander";
+  version = "011500";
```

Comment:
You are sure the version number isn't 1.15.0?

Reply:
the relase page states no version number but a tag called `libopenblt_011500`
the problem is that this is not the actual version of the application but the version of the repo that contains both the library and the bootcommander application. Therefore I don't know how to set the correct version of the application and library and use that variable to fetch the repo since it differs to the github tag

Reply:
and that version in the tag does not translate to anything? then we just use that, because it is easier

Reply:
Seems like official versions use dots: https://sourceforge.net/projects/openblt/files/OpenBLT%20stable/

Reply:
on svn/sourceforge yes, but on github no, so you would need a different value for pulling from GH and one for assigning the version

Reply:
To me it looks like versions and tags correspond:

![image](https://github.com/NixOS/nixpkgs/assets/8182846/83c79712-87bc-40cf-ae61-c30b3a746663)
![image](https://github.com/NixOS/nixpkgs/assets/8182846/88d375be-9e6b-44da-932a-8f05ae252c44)

so this should work:

```nix
version = "1.16.00";
src = fetchFromGitHub {
  rev = "openblt_v0${lib.replaceStrings ["."] [""] version}";
};
```

Thechincally this will break as soon as version 10.x.y is released, but I think it's good enough (I don't see any padding function in lib/)

Reply:
Is this really necessary? Since we already know that it will break and the value by itself is not very usefull because the software has a different version, can't we keep it simple like this? Moreover will automatic updates handled by ryantm support this?

Reply:
> Is this really necessary?

Yes, the `version` attribute should reflect the actual version. Tools such as repology rely on it. There are [many cases like this one](https://github.com/search?q=repo%3ANixOS%2Fnixpkgs+%22rev+%3D%22+replaceStrings&type=code).

> Since we already know that it will break

It will break... in 10 major versions, which won't happen any time soon. You can make it more robust by conditionally adding that zero if you'd like to.

> Moreover will automatic updates handled by ryantm support this?

I think so. It can get notified by repology and the revision will is derived from the version.

Reply:
https://nixos.org/manual/nixpkgs/stable/index.html#var-passthru-updateScript

Reply:
done